src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -167,25 +169,113 @@ public static unsafe void MatMulX(bool add, AlignedArray mat, AlignedArray src,
                     Vector256<float> res2 = res0;
```

Comment:
This code is currently generating:
```
vxorps   ymm0, ymm0, ymm0
vmovupd  ymmword ptr[rsp+40H], ymm0
vmovupd  ymm6, ymmword ptr[rsp+40H]
vmovupd  ymm7, ymmword ptr[rsp+40H]
vmovupd  ymm8, ymmword ptr[rsp+40H]
```

Which, while not a "likely" bottleneck, is still less than stellar.

Reply:
Changing this to just call `Avx.SetZeroVector256<float>()` multiple times improves the codegen.

Reply:
should we do similar thing for SseIntrinsics  ?

Reply:
Yes.